app/routes/events/components/EventAttendeeStatistics.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+
+import { PieChart, Pie, Cell } from 'recharts';
+import { CHART_COLORS } from 'app/routes/surveys/utils';
+import type { EventRegistration } from 'app/models';
+import { renderCustomizedLabel } from 'app/routes/surveys/components/Submissions/Results';
+import { Flex } from 'app/components/Layout';
+
+interface DistributionDataPoint {
+  name: string;
+  count: number;
+}
+
+const DistributionPieChart = ({
+  distributionData,
+}: {
+  distributionData: DistributionDataPoint[],
+}) => {
+  return (
+    <Flex alignItems="center">
+      <PieChart width={400} height={350}>
+        <Pie
+          data={distributionData}
+          cx={200}
+          cy={150}
+          labelLine={false}
+          label={renderCustomizedLabel}
+          dataKey="count"
+        >
+          {distributionData.map((entry, index) => (
+            <Cell
+              key={`cell-${index}`}
+              fill={CHART_COLORS[index % CHART_COLORS.length]}
+            />
+          ))}
+        </Pie>
+      </PieChart>
+      <div>
+        <ul>
+          {distributionData.map((dataPoint, i) => (
+            <li key={i}>
+              <span
+                style={{
+                  backgroundColor: CHART_COLORS[i % CHART_COLORS.length],
+                }}
+              >
+                &nbsp;
+              </span>
+              <span> {dataPoint.name}</span>
+            </li>
+          ))}
+        </ul>
+      </div>
+    </Flex>
+  );
+};
```

Comment:
I love that the pie chart is a separate component and abstract enough to be used for other things as well! 

I feel like it should be moved into a separate file in /components, and possibly also be reused in the survey result page, as that chart seems to be very similar.

Reply:
Actually, while you're at it you should probably move the bar chart from survey results into a separate component as well. No reason to have all this chart-creation code inside the survey route folder if we also use charts outside of surveys.

Reply:
Yeah, I agree! ✅